Python/bytecodes.c

```diff
@@ -3765,13 +3765,13 @@ dummy_func(
             DEOPT_IF(!PyType_Check(callable_o));
             PyTypeObject *tp = (PyTypeObject *)callable_o;
             DEOPT_IF(FT_ATOMIC_LOAD_UINT32_RELAXED(tp->tp_version_tag) != type_version);
-            assert(tp->tp_flags & Py_TPFLAGS_INLINE_VALUES);
+            assert(tp->tp_flags & Py_TPFLAGS_HEAPTYPE);
```

Comment:
Maybe assert `tp->tp_new == PyBaseObject_Type.tp_new` and `tp->tp_alloc == PyType_GenericAlloc`